packages/ui/src/context/CartProvider.tsx

```diff
@@ -359,8 +465,42 @@ function cartStore({
     commit()
   }, [commit])
 
+  const setQuantity = useCallback(
+    (orderId: string, quantity: number) => {
+      const updatedItems = [...cartData.current.items]
+      let item = updatedItems.find((item) => item.order?.id === orderId)
```

Comment:
we should probably also check that quantity remaining is not > the quantity passed in. If it is just set it to the max.